tree.go

```diff
@@ -301,10 +301,13 @@ func NewLeafNodeWithNoComms(stem []byte, values [][]byte) *LeafNode {
 	}
 }
 
+// Children return the children of the node. The returned slice is
+// internal to the tree, so callers *must* consider it readonly.
 func (n *InternalNode) Children() []VerkleNode {
 	return n.children
 }
 
+// SetChild *replaces* the child at the given index with the given node.
 func (n *InternalNode) SetChild(i int, c VerkleNode) error {
 	if i >= NodeWidth {
 		return errors.New("child index higher than node width")
```

Comment:
Both `Children()` and `SetChild()` aren't methods used in `go-verkle`. Tracking down where they were used, they're necessary for the iterator-API in `go-ethereum`.

Added some comments to warn callers that should be mindful about what they're doing with returned parameters, since external parties touching these things can break some invariants in this package.